numpyro/distributions/util.py

```diff
@@ -1,24 +1,20 @@
 from functools import update_wrapper
 import math
-from numbers import Number
 
 import scipy.special as osp_special
 
-from jax import custom_transforms, defjvp, device_get, jit, lax, random, vmap
+from jax import custom_transforms, defjvp, jit, lax, random, vmap
 from jax.lib.xla_bridge import canonicalize_dtype
 import jax.numpy as np
 from jax.numpy.lax_numpy import _promote_args_like
 from jax.scipy.linalg import solve_triangular
 from jax.util import partial
 
 
-# TODO: inefficient implementation; jit currently fails due to
-# dynamic size of random.uniform.
-@partial(jit, static_argnums=(2, 3))
-def _binomial(key, p, n, shape):
+@partial(jit, static_argnums=(3, 4))
+def _binomial(key, p, n, n_max, shape):
```

Comment:
Good idea to jit n_max instead of n!

Reply:
Yes, it helps a lot! But I still can't find a way to make this works for a traced array `n`. I believe that we should revise the algorithm a bit for that purpose. But it is not worth to think about this now.